src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -663,14 +671,8 @@ private void finish(@NonNull Result r, @CheckForNull Throwable t) {
             onEndBuilding();
         } finally {  // Ensure this is ALWAYS removed from FlowExecutionList
             FlowExecutionList.get().unregister(new Owner(this));
-            listener = null;
-        }
-
-        try {
-            StashManager.maybeClearAll(this, /* or move up before closing getListener()? */new LogTaskListener(LOGGER, Level.FINE));
```

Comment:
Fixes a longstanding issue, that for example https://github.com/jenkinsci/artifact-manager-s3-plugin/blob/b8c98c6978ea3d17517e936b73ac4e98b735e934/src/main/java/io/jenkins/plugins/artifact_manager_jclouds/JCloudsArtifactManager.java#L355 would not be seen.